doc/design/linc_permissions.md

```diff
@@ -19,25 +34,25 @@ https://github.com/google/neuroglancer/issues/507
 
 ## Implementation Options
 
```

Comment:
Moved from below into a new `Implemented solution` section.

```suggestion
## Implemented solution

- S3 Asset Bucket is served via a CloudFront distribution

- User navigates to LINC Archive (i.e. `lincbrain.org`) to authenticate. Thereby the user hits a get_presigned_cookie/ API endpoint. In the endpoint response is a valid cookie that allows the user's browser to be able to access the asset via their browser.

Render with hosted Neuroglancer
- Host Neuroglancer fork on S3 bucket with static site hosting enabled. Create an AWS CloudFront distribution to sit as a verified proxy for permission to many S3 assets at once via presigned cookies.  Provide CNAME for CloudFront Distribution URL to `neuroglancer.lincbrain.org` so that cookies generated by LINC Archive (also living at a `*.lincbrain.org`) can be shared between the API and the Neuroglancer site.
- Modify the Neuroglancer source code to handle a CloudFront asset path (i.e. `neuroglancer.lincbrain.org` since the domains of the presigned cookie are limited to `*.lincbrain.org` domains due to same-site requirements for cookies) that is similar to any S3 asset path (i.e. `s3://`).
- Cookies would get passed in request to get asset

Access assets with LINC CLI
  - Cookies would get passed in request to get asset
  - Provide LINC CLI option for retrieving asset. Instead of just referencing the asset directly, we would need to provide a helper function that wraps the request with the cookie.

Architecture diagram
```mermaid
%%{init: {"flowchart": {"curve": "linear"}}}%%
flowchart LR
    E --> B
    A(User) -->| If client has CloudFront cookies from prior session, <br/> then proceed. | B(Static Webpage i.e. Neuroglancer)
    A --> | If client does not have CloudFront cookies, <br/> then GET upon /api/permissions/s3 in LINC Archive API. | E(LINC Archive API)
    B --> | Upon user activity, <br/> sends presigned cookies | C(AWS CloudFront)
    C -->| Allows data to be fetched | D(Private AWS S3 Bucket)
    D -->| 1. Neuroglancer able to access S3 data <br/> 2. Data rendered on screen | B
```

## Alternative options

```